File: 9-palindrome-number/palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x<0:
            return False
        elif x>=0 and x<=9:
            return True
        elif not x%10:
            return False
        rev=0
        num=x
        while x:
            rev*=10
            rev+=x%10
            x//=10
        return rev==num
        

File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0)return false;
        int num=x;
        int rev=0;
        while(x){
            if(rev>INT_MAX/10){
                return false;
            }
            rev=rev*10+x%10;
            x/=10;
        } 
        if(rev==num){
            return true;
        } 
        return false;
    }
};